<?xml version="1.0"?>
<launch>
    <arg name="start_pt" default= '-8.01, -16.2, 0.9877181, 0.0'/>
    <arg name="use_rviz" default="true"/>
    <arg name="is_sim" default="false"/>
    <arg name="use_camera" default="true"/>

    <arg name="map_file" default="$(find svea_core)/resources/maps/floor2.yaml" />

    <group if="$(eval is_sim == true)">
	<node pkg="map_server" type="map_server" name="map_server" output="screen" args="$(arg map_file)" /> 
    <!--node type="rviz" name="rviz" pkg="rviz" args="-d $(find svea_core)/resources/rviz/c_ovrv.rviz" /-->
    </group>

    <group if="$(eval use_camera == true)">
        <include file="$(find svea_core)/launch/vlac_marker_detection.launch"/>
    </group> 

    <group if="$(eval is_sim == false)">
        <!--open serial connection for controlling SVEA-->
	    <node pkg="rosserial_python" type="serial_node.py" name="serial_node">
    	    <param name="port" value="/dev/ttyACM0"/>
    	    <param name="baud" value="250000"/>
	    </node>
	    <include file="$(find svea_sensors)/launch/localize.launch">
    	    <arg name="use_rs" value="true"/>
    	    <arg name="file_name" value="$(arg map_file)"/>			
	    </include>
    </group>

    <node pkg="svea_core" type="vision_MPC.py" name="vision_MPC" output="screen">
        <param name="is_sim" value="$(arg is_sim)"/>
        <param name="use_camera" value="$(arg use_camera)"/>
    </node>
 
    <node pkg="svea_core" type="c_ovrv_MPC.py" name="c_ovrv_MPC" output="screen">
        <param name="start_pt" value="$(arg start_pt)"/>
        <param name="use_rviz" value="$(arg use_rviz)"/>
        <param name="is_sim" value="$(arg is_sim)"/>
        <param name="use_camera" value="$(arg use_camera)"/>
    </node>
    

</launch>
